server/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java

```diff
@@ -73,20 +74,23 @@ private static void parseAnalyzersAndTermVectors(FieldMapper.Builder builder, St
                 if (analyzer == null) {
                     throw new MapperParsingException("analyzer [" + propNode.toString() + "] not found for field [" + name + "]");
                 }
+                analyzer.checkAllowedInMode(AnalysisMode.ALL);
```

Comment:
Should we check the `AnalysisMode.INDEX_TIME` mode if `search_analyzer` is also set ?

Reply:
Just to clarify: you mean we should allow an "analyzer" to be restricted to "index time" only if the search analyzer is set?

Reply:
I think so yes, if the analyzer is set to INDEX_TIME this means that it should not be used for queries so we should check that. My understanding is that the ALL mode check will fail if the analyzer is set to INDEX_TIME even though a search analyzer is defined, this looks like a bug to me.

Reply:
Yes, that makes sense. Good point, will add that.